src/snowflake/snowpark/dataframe.py

```diff
@@ -5941,27 +5942,61 @@ def random_split(
                     raise ValueError("weights must be positive numbers")
 
             temp_column_name = random_name_for_temp_object(TempObjectType.COLUMN)
-            cached_df = self.with_column(
-                temp_column_name,
-                abs_(random(seed)) % _ONE_MILLION,
-                _emit_ast=False,
-            ).cache_result(statement_params=statement_params, _emit_ast=False)
-            sum_weights = sum(weights)
-            normalized_cum_weights = [0] + [
-                int(w * _ONE_MILLION)
-                for w in list(itertools.accumulate([w / sum_weights for w in weights]))
-            ]
-            normalized_boundaries = zip(
-                normalized_cum_weights[:-1], normalized_cum_weights[1:]
-            )
-            res_dfs = [
-                cached_df.where(
-                    (col(temp_column_name) >= lower_bound)
-                    & (col(temp_column_name) < upper_bound),
-                    _emit_ast=False,
-                ).drop(temp_column_name, _emit_ast=False)
-                for lower_bound, upper_bound in normalized_boundaries
-            ]
+            with ResourceUsageCollector() as resource_usage_collector:
+                if self._session.conf.get("use_simplified_query_generation"):
+                    api_name = "DataFrame.random_split[hash]"
+                    if seed is not None:
+                        local_random = random.Random(seed)
+                        python_seed = local_random.random()
+                    else:
+                        python_seed = random.random()
+                    intermediate_df = self.with_column(
+                        temp_column_name,
+                        abs_(
+                            hash_(
+                                "*", lit(python_seed, _emit_ast=False), _emit_ast=False
+                            ),
```

Comment:
idea here is to hash all column for determinism and use `python_seed` as a source of randomness. When `seed` is provided as an input, `python_seed` will repeatably produce the same random number.